### ISSUE TITLE
Input prior inconsistent with initial_state_prior (sts.LocalLevel)

### ISSUE THREAD

**Issue by howardya** [tensorflow/probability#1481]
This might turn out to clarification, but feel free to let me know and I can close it.

In the simplest cases of 
[LocalLevel](https://github.com/tensorflow/probability/blob/v0.15.0/tensorflow_probability/python/sts/components/local_level.py#L302-L311)
[LocalLinearTrend](https://github.com/tensorflow/probability/blob/v0.15.0/tensorflow_probability/python/sts/components/local_linear_trend.py#L382-L390)

the highlighted lines above define initial_state_prior through a MultivariateNormalDiag with mean and scale defined by initial_level_prior or initial_slope_prior.

May I know why are the priors defined as MultivariateNormalDiag even though the initial_level_prior/initial_scale_prior inputs may not be Normal? 

For example, suppose I have the following simple random walk model, where my initial_level_prior is a LogNormal distribution.
```
trend = sts.LocalLevel(
    observed_time_series = co2_by_month_training_data,
    initial_level_prior=tfd.LogNormal(
        loc=2.,
        scale=1.
    )
)
```

If I check the initial_state_prior, as expected, the prior has been redefined as MultiNormal (which in this is simply a normal distribution)
```
trend.initial_state_prior
```
`
<tfp.distributions.MultivariateNormalDiag 'MultivariateNormalDiag' batch_shape=[] event_shape=[1] dtype=float32>
`

My questions are:
1. What is the rationale for overwriting the initial_state_prior with MultivariateNormalDiag as opposed to a joint independent distribution?
2. Is there a way for me to enforce the initial_state_prior to user defined distribution?

Thank you.

**Comment by junpenglao** [comment]
I think there is a miss-conception here that is best to be clarified with a picture (this is from `Martin, O. A., Kumar, R., & Lao, J. (2021). Bayesian Modeling and Computation in Python`):
![fig18_bsts_lgssm](https://user-images.githubusercontent.com/12952641/146634923-6847717e-4356-4323-9611-d81b29bb78f5.png)

Per the formulation of linear Gaussian State space model, latent states (X in the figure) need to be Gaussian, otherwise you cannot use Kalman filter to do compute log_prob and do state update etc.

I think maybe restricting the initial_level_prior/initial_scale_prior to `tfd.Normal` could make things clearer, but there could be design choice or other places using this information that I am not aware of. @davmre probably could provide a bit more information.

**Comment by howardya** [comment]
@junpenglao Thank you for your insight. Now I understand a little more. The variables _initial_state_prior or self.initial_state_prior are indeed meant for state space model. They are not used for VI.

For instance, in the example in the [Jupiter](https://github.com/tensorflow/probability/blob/v0.15.0/tensorflow_probability/examples/jupyter_notebooks/Structural_Time_Series_Modeling_Case_Studies_Atmospheric_CO2_and_Electricity_Demand.ipynb), initial_state_prior is not used at all. The function

```
variational_posteriors = tfp.sts.build_factored_surrogate_posterior(
    model=co2_model)
```

build the prior using [co2_model._joint_prior_distribution](https://github.com/tensorflow/probability/blob/v0.15.0/tensorflow_probability/python/sts/structural_time_series.py#L243-L254) which is indeed using the initial_prior defined by users.

In summary:
For VI and non-State Space Model analysis, the priors are consistent with user defined initial_level_prior/initial_scale_prior. 
For SSM, the latent state models are using MultivariateNormalDiag.

Appreciate if someone can confirm my above understanding and if this will be changed in the future.

**Comment by junpenglao** [comment]
> For VI and non-State Space Model analysis, the priors are consistent with user defined initial_level_prior/initial_scale_prior.
> For SSM, the latent state models are using MultivariateNormalDiag.

If by "non-State Space Model analysis" you meant inferring the hyper-parameter (the most obvious one being the sigma of the SSM) of a SSM, then yes.
